WalletWasabi/Crypto/ZeroKnowledge/ZkVerifier.cs

```diff
@@ -0,0 +1,25 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class ZkVerifier
+	{
+		public static bool Verify(ZkExponentProof proof)
+		{
+			Guard.NotNull(nameof(proof), proof);
+
+			var publicPoint = proof.PublicPoint;
```

Comment:
This is wrong. What you call `PublicPoint` here (the prover's public key) is something that has to be known by the Verifier in advance and it is what is used to verify that the proof is correct. How could you know that a proof is valid if you have nothing to compare with? Impossible.

For example, I know the Wasabi Wallet's creator public key is `abcdef1234....` then someone asks me for the wasabi server credentials claiming he is the Wasabi Wallet's creator and he forgot the password. I can ask that person to prove he is who he says by sending me a valid signature (proof of exponent) and I (the Verifier) can verify the proof is correct because I know the public key. Obviously I cannot trust on him if he sends me the signature and also the public key that I have to use during the verification process. **Disclaimer:** I can check that the public key that the prover sent to me matches exactly with the one I already know/have but conceptually this is wrong to me.